### PR TITLE
Add flag to pass user auth tokens at runtime

### PR DESCRIPTION
**Pull request by noorvir**
- Adds `--token` flag to optionally pass a user auth token with every command. This override the login token
- Adds new `token` command with `get-user-tokens` and `create-user-token` sub commands to configure auth tokens for the current user.

Resolves UNW-169

### CODE REVIEW

File: cmd/token.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	for _, token := range tokens {
+		fmt.Printf("\t%s\n", token)
+	}
+	return nil
+}
+
+func CreateUserTokenCmd(cmd *cobra.Command, args []string) error {
+	h := New()
+	ctx := context.Background()
```
Comment (igomez10) [inline cmd/token.go:35]: To make testing easier we can pass this context and handler from the function caller as arguments

File: cmd/token.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	cmd.SilenceUsage = true
+	return h.CreateUserToken(ctx, &entity.Command{
+		Args: args,
+		Cmd:  cmd,
+	})
+}
+
+func GetUserTokensCmd(cmd *cobra.Command, args []string) error {
+	h := New()
+	ctx := context.Background()
```
Comment (igomez10) [inline cmd/token.go:45]: same here

File: constants/env.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
 package constants
 
 var UnweaveEnv = "dev"
-
+var AuthToken = ""
```
Comment (igomez10) [inline constants/env.go:4]: Please change from var to const and change to uppercase. Given the name of the file I imagine these are constants

File: entity/token.go [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
+	ExpiresAt   int64  `json:"expiresAt"`
 	Token       string `json:"token"`
 }
 
 func (u UserToken) String() string {
-	return fmt.Sprintf("ID: %s, DisplayName: %s, ExpiresAt: %d", u.ID, u.DisplayName, u.ExpiresAt)
+	expires := time.UnixMilli(u.ExpiresAt)
+	expiresStr := expires.Format("2006-01-02 15:04:05")
+	return fmt.Sprintf("ID: %s, DisplayName: %s, ExpiresAt: %s", u.ID, u.DisplayName, expiresStr)
 }
```
Comment (igomez10) [inline entity/token.go:20]: Please use type `time.Time` for `expiresAt`. Doing the date conversion only in String() does make a lot of sense, we can define types for the GraphQL response `GetUserTokenResponse` and an intermediate function that converts `GetUserTokenResponse` to `UserToken`.

File: main.go [hunk 5]
```diff
@@ -91,6 +94,30 @@ func init() {
...
+	}
+	token.AddCommand(getUserTokens)
+
+	createUserToken := &cobra.Command{
+		Use:   "create-user-token",
+		Short: "Create a new token",
+		RunE:  cmd.CreateUserTokenCmd,
+	}
+	token.AddCommand(createUserToken)
+	rootCmd.AddCommand(token)
```
[the remark this answers is not in the document]
Comment (noorvir) [reply main.go]: Done ✅

File: cmd/token.go (hunk 1 above)
Comment (noorvir) [reply cmd/token.go:35]: This is the entry point for the command. I don't expect we'll ever be testing this function. This just setup up the context for calling a handler. If we want to test the functionality of a command, you would replace this function with something like:

```go
func SetupTestUserTokenCmd() error {
	h := NewTestHandler() 
	ctx := context.Background()

	return h.CreateUserToken(ctx, &entity.Command{
		Args: []string{"arg1", "arg2"},
		Cmd:  &cobra.Command{...},
	})
}

```

File: entity/token.go (hunk 4 above)
Comment (noorvir) [reply entity/token.go]: Good idea! Updated this with a custom `UnmarshalJSON`

**Pull request merged**